.github/workflows/publish.yml

```
@@ -0,0 +1,64 @@
+name: Publish
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+    paths:
+      - 'package.json'
+
+permissions:
+  id-token: write
+  contents: read
+
+env:
+  CDN_BUCKET: gc-design-system-production-cdn
+  CDN_REGION: ca-central-1
+  PACKAGE_NAME: '@cdssnc/gcds-fonts'
+
+jobs:
+  build-deploy:
+    name: Publish
+    runs-on: ubuntu-latest
+    steps:
+      - name: Git Checkout
+        uses: actions/checkout@eef61447b9ff4aafe5dcd4e0bbf5d482be7e7871 # v4.2.1
+
+      - name: Publish @cdssnc/gcds-fonts to npm
+        uses: JS-DevTools/npm-publish@e06fe3ef65499b38eb12224f2a60979f6d797330
+        id: publish
+        with:
+          token: ${{ secrets.NPM_TOKEN }}
+
```

Should we add a sleep to this workflow so we don't get the problems we sometimes have on other repos?

```suggestion

      - name: Sleep for 60 seconds to give more time for NPM to complete publishing before proceeding to publish the rest
        run: sleep 60
        
```

---

Great idea! Or it would be even better to separate the CDN upload (with a wait check if the package has been published), I think we have something like that on the components package (or maybe tokens / utility). We've ran into an issue before where we wanted to just upload it to the CDN, the publishing portion was done, but it can't proceed to the CDN upload because it relied on the publish output.

---

Great idea Ethan! We'll probably be ok with Ethan's suggestion for the fonts repo. That's what we are using in the tokens repo and in the utility repo, we don't have anything added at all and never had any issues. We may have more problems with the components repo because it's a lot of different packages that need to be generated and published on NPM.

---

It would be ideal to move the tokens and utility to have the upload cdn in a separate file, it would be nice to start doing it for new repos instead of adding more tech debt. The NPM publish action can get stuck when npm has issues (we've seen it before) no matter how big the package is. It doesn't happen often, but it does and it's a pain point

---

Here's an incident for the tokens that happened just last December:
https://docs.google.com/document/d/1ZYBqG64CMcojX27Q6gggsIEPXzXGq8bHcBdzvNLCznM/edit?tab=t.0#heading=h.gjdgxs

---

I can make the changes if you like, I was just trying to follow the "done is better than perfect" approach we seem to be using more lately as this is a very low-profile repo that will hardly ever get changed.

---

When an incident happens, this will need to be fixed. So yes, done is better than perfect but we need to recognize that one of us will need to fix this as has already happened before, and if it does happen, you will need to call it an incident and it will be an emergency fix then when it is avoidable right now.